app/src/main/java/app/service/service/ServiceService.java

```diff
@@ -117,56 +133,65 @@ public ServiceDTO updateService(Long serviceId, UpdateServiceDTO serviceDTO, Str
 
         Optional<Service> serviceOptional = serviceRepository.findByIdAndJurisdictionId(serviceId, jurisdictionId);
         if (serviceOptional.isEmpty()) {
-            LOG.error("Service not found.");
-            return null;
+           throw new ServiceNotFoundException(serviceId, jurisdictionId);
         }
 
         Service service = serviceOptional.get();
         if (serviceDTO.getServiceCode() != null) {
-            if (serviceCodeAlreadyExists(serviceDTO.getServiceCode(), service.getJurisdiction())) {
-                return null;
-            }
+            validateServiceCodeExistence(serviceDTO.getServiceCode(), service.getJurisdiction());
             service.setServiceCode(serviceDTO.getServiceCode());
         }
 
         if (serviceDTO.getGroupId() != null) {
-            if (groupDoesNotExists(serviceDTO.getGroupId(), service.getJurisdiction(), service)) return null;
+            ServiceGroup group = validateGroupExistenceAndReturn(serviceDTO.getGroupId(), jurisdictionId);
+            service.setServiceGroup(group);
         }
-
         if (serviceDTO.getDescription() != null) {
             service.setDescription(serviceDTO.getDescription());
         }
         if (serviceDTO.getServiceName() != null) {
             service.setServiceName(serviceDTO.getServiceName());
         }
+        if (serviceDTO.getOrderPosition() != null) {
+            service.setOrderPosition(serviceDTO.getOrderPosition());
+        }
 
         return toServiceDTO(serviceRepository.update(service));
     }
 
+    public List<ServiceDTO> updateServiceOrderPositions(Long groupId, List<PatchServiceOrderPositionDTO> requestDTO) {
+
+        updateServicesOrderPosition(groupId, requestDTO);
+
+        // get refreshed list of services
+        List<Service> services = serviceRepository.findAllByServiceGroupId(groupId);
```

Comment:
The backend is in charge of ordering the items so here we need to use `OrderBy`

Reply:
Good call out, thank you! I'll make the update.